src/v/redpanda/application.cc

```diff
@@ -317,23 +316,56 @@ void application::initialize(
 }
 
 void application::setup_metrics() {
-    if (!config::shard_local_cfg().disable_public_metrics()) {
-        seastar::metrics::replicate_metric_families(
-          seastar::metrics::default_handle(),
-          {{"scheduler_runtime_ms", ssx::metrics::public_metrics_handle},
-           {"io_queue_total_read_ops", ssx::metrics::public_metrics_handle},
-           {"io_queue_total_write_ops", ssx::metrics::public_metrics_handle},
-           {"memory_allocated_memory", ssx::metrics::public_metrics_handle},
-           {"memory_free_memory", ssx::metrics::public_metrics_handle}})
-          .get();
-    }
+    setup_internal_metrics();
+    setup_public_metrics();
+}
 
-    if (config::shard_local_cfg().disable_metrics()) {
+void application::setup_public_metrics() {
+    namespace sm = ss::metrics;
+
+    if (config::shard_local_cfg().disable_public_metrics()) {
         return;
     }
 
+    seastar::metrics::replicate_metric_families(
+      seastar::metrics::default_handle(),
+      {{"scheduler_runtime_ms", ssx::metrics::public_metrics_handle},
+       {"io_queue_total_read_ops", ssx::metrics::public_metrics_handle},
+       {"io_queue_total_write_ops", ssx::metrics::public_metrics_handle},
+       {"memory_allocated_memory", ssx::metrics::public_metrics_handle},
+       {"memory_free_memory", ssx::metrics::public_metrics_handle}})
+      .get();
+
+    _public_metrics.add_group(
+      "application",
+      {
+        sm::make_gauge(
+          "uptime_seconds_total",
+          [] {
+              return std::chrono::duration<double>(ss::engine().uptime())
+                .count();
+          },
+          sm::description("Redpanda uptime in seconds"))
+          .aggregate({sm::shard_label}),
+        sm::make_gauge(
+          "busy_seconds_total",
+          [] {
+              return std::chrono::duration<double>(
+                       ss::engine().total_busy_time())
+                .count();
+          },
+          sm::description("Total CPU busy time in seconds"))
+          .aggregate({sm::shard_label}),
```

Comment:
I think it's worth keeping the shards here

Reply:
These metrics are only reported from one shard. That's why I aggregated.

Reply:
```
curl -s  localhost:19644/metrics | grep  cpu_busy_ms
# HELP vectorized_reactor_cpu_busy_ms Total cpu busy time in milliseconds
# TYPE vectorized_reactor_cpu_busy_ms counter
vectorized_reactor_cpu_busy_ms{shard="0"} 6268
vectorized_reactor_cpu_busy_ms{shard="1"} 5003
```

Reply:
Oh, you mean this probe is only on one shard? Can you register a metric for each shard?

Reply:
Right. I've confused myself. It's reported from one shard because it's only registered on one shard. We should probably register it on all shards. This means that's probably not the right place for the "busy_time" metric. Let me have a think.

Reply:
You can `invoke_on`, or `submit_to`, or create a `sharded<probe>` and `.invoke_on_all`, or move it.

Reply:
I know. Just feels a bit unnatural to do it there.

Reply:
Btw. Do we want the uptime for every shard? It would probably be redundant. Busy time for each shard makes sense though.

Reply:
Yep, that's how it is:
```
curl -s localhost:19644/metrics | grep uptime
# HELP vectorized_application_uptime Redpanda uptime in milliseconds
# TYPE vectorized_application_uptime gauge
vectorized_application_uptime{shard="0"} 24680.000000
```

Reply:
Done. I wrapped the metric_groups object into a `ss::sharded`. `uptime` is only reported from the home shard, while `busy` is reported from all shards.